packages/lesswrong/client/scrollRestoration.ts

```diff
@@ -1,17 +1,22 @@
+import { commentPermalinkStyleSetting } from "@/lib/publicSettings";
 
 /**
  * When refreshing the page, tell the browser to remember the scroll position.
  * Otherwise, users get scrolled to the top of the page.
  * (See https://github.com/Lesswrong2/Lesswrong2/issues/295#issuecomment-385866050)
  */
 export function rememberScrollPositionOnPageReload() {
-  window.addEventListener('beforeunload', () => {
-    if ('scrollRestoration' in window.history) {
+  window.addEventListener("beforeunload", () => {
+    const urlParams = new URLSearchParams(window.location.search);
+    const commentId = urlParams.get("commentId");
```

Comment:
I deliberately decided to not handle the hash case here, because hashes can be used for other things. This means the browser scroll restoration might still fight the hash version, if the page is v slow to load